releases_test.go

```diff
@@ -22,6 +23,86 @@ type kustomizationFile struct {
 	Transformers      []string          `yaml:"transformers"`
 }
 
+// requestException represents a single release exception to a request
+type requestException struct {
+	Version string `yaml:"releaseVersion" json:"releaseVersion"`
+	Reason  string `yaml:"reason"`
+}
+
+// versionRequest represents a specific requested component name and version
+type versionRequest struct {
+	Name       string             `yaml:"name"`
+	Version    string             `yaml:"version"`
+	Exceptions []requestException `yaml:"except,omitempty" json:"except,omitempty"`
+}
+
+// releaseRequest is one release pattern with associated requests
+type releaseRequest struct {
+	Name     string           `yaml:"name"`
+	Requests []versionRequest `yaml:"requests"`
+}
+
+type requestsFile struct {
+	Releases []releaseRequest `yaml:"releases"`
+}
+
+// componentListSatisfiesRequest determines whether the given request is satisfied in the given component list.
+// It returns a boolean value for whether the request is satisfied as well as
+// a string containing the actual component version which satisfies the request.
+func componentListSatisfiesRequest(request versionRequest, componentList []v1alpha1.ReleaseSpecComponent) (bool, string, error) {
+	var actual string
+	for _, component := range componentList {
+		if component.Name == request.Name {
+			actual = component.Version
+			actualMatchesRequested, err := versionMatches(actual, request.Version)
+			if err != nil {
+				return false, actual, microerror.Mask(err)
+			}
+
+			if actualMatchesRequested {
+				return true, actual, nil
+			}
+
+			break // No need to keep searching for this component
+		}
+	}
+	return false, actual, nil
+}
+
+// findMatchingRequests searches the given array of releaseRequests
+// for requests which apply to the given release version
+func findMatchingRequests(release string, requests []releaseRequest) ([]versionRequest, error) {
+	var requestList []versionRequest
+	for _, request := range requests {
+
+		// See whether this request applies to the current release version
+		match, err := versionMatches(release, request.Name)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		if match {
+			for _, component := range request.Requests {
+				releaseIsExcluded := false
+				if component.Exceptions != nil {
+					// Check the excluded releases for this component to see if our release is there
```

Comment:
Let's add `.` at the end of all comments at least kind of consistently :)

Reply:
Just for you